13-Tenant Security Solution Setup/Scripts/RemediationScripts/Remediate-ConfigASCTier.ps1

```diff
@@ -0,0 +1,389 @@
+        }
+        catch 
+        {
+            Write-Host "Error occured while checking pre-requisites. ErrorMessage [$($_)]" -ForegroundColor Red    
+            break
+        }
+    }
+
+    # Connect to AzAccount
+    $isContextSet = Get-AzContext
+    if ([string]::IsNullOrEmpty($isContextSet))
+    {       
+        Write-Host "Connecting to AzAccount..."
+        Connect-AzAccount
+        Write-Host "Connected to AzAccount" -ForegroundColor Green
+    }
+
+    # Setting context for current subscription.
+    $currentSub = Set-AzContext -SubscriptionId $SubscriptionId
+
+    
+
+    Write-Host "Metadata Details: `n SubscriptionId: [$($SubscriptionId)] `n AccountName: [$($currentSub.Account.Id)] `n AccountType: [$($currentSub.Account.Type)]"
+    Write-Host "------------------------------------------------------"
+    Write-Host "Starting with Subscription [$($SubscriptionId)]..."
+
+
+    Write-Host "Step 1 of 3: Validating whether the current user [$($currentSub.Account.Id)] have the required permissions to run the script for Subscription [$($SubscriptionId)]..."
+
+    # Safe Check: Checking whether the current account is of type User
+    if($currentSub.Account.Type -ne "User")
+    {
+        Write-Host "Warning: This script can only be run by user account type." -ForegroundColor Yellow
+        break;
+    }
+
+    # Safe Check: Current user need to be either Contributor or Owner for the subscription
+    $currentLoginRoleAssignments = Get-AzRoleAssignment -SignInName $currentSub.Account.Id -Scope "/subscriptions/$($SubscriptionId)";
+
+    if(($currentLoginRoleAssignments | Where { $_.RoleDefinitionName -eq "Owner"  -or $_.RoleDefinitionName -eq 'Contributor' } | Measure-Object).Count -le 0)
+    {
+        Write-Host "Warning: This script can only be run by an Owner or Contributor of subscription [$($SubscriptionId)] " -ForegroundColor Yellow
+        break;
+    }
+
+    Write-Host "`n"
+    Write-Host "Step 2 of 3: Fetching remediation log to perform roll back operation to config ASC tier for Subscription [$($SubscriptionId)]..."
+ 
+    # Array to store resource context
+    if (-not (Test-Path -Path $Path))
+    {
+        Write-Host "Warning: Rollback file is not found. Please check if the initial Remediation script has been run from the same machine. Exiting the process" -ForegroundColor Yellow
+        break;        
+    }
+
+    # Declaring required ASC type and pricing tier
+    $reqASCTier = "Standard";
+    $reqProviderName = "Microsoft.Security"
+    $remediatedLog = Get-Content -Raw -Path $Path | ConvertFrom-Json
+
+    Write-Host "Step 3 of 3: Performing roll back operation to config ASC tier for subscription [$($SubscriptionId)]..."
+        
+    # Performing roll back operation
+    try
+    {
+        if(($remediatedLog | Measure-Object).Count -gt 0)
+        {
+            Write-Host "Configuring ASC tier as per remediation log on subscription [$($SubscriptionId)]..."
+            
```

Comment:
here we are not remediating, we are doing rollback.

Reply:
Correct